### PR TITLE
Port RSA to rust

### PR DESCRIPTION
**Pull request by alex**
Has a number of TODOs that need to be dealt with

### CODE REVIEW

File: src/rust/src/backend/ec.rs [hunk 1]
```diff
@@ -503,10 +503,6 @@ impl ECPublicKey {
         let mut verifier = openssl::pkey_ctx::PkeyCtx::new(&self.pkey)?;
         verifier.verify_init()?;
         let valid = verifier.verify(data, signature).unwrap_or(false);
-        // TODO: Empty the error stack. BoringSSL leaves one in the event of
-        // signature validation failure. Upstream to rust-openssl?
-        #[cfg(CRYPTOGRAPHY_IS_BORINGSSL)]
```
Comment (reaperhulk) [inline src/rust/src/backend/ec.rs:508]: Should this be a separate PR?

Comment (alex) [reply src/rust/src/backend/ec.rs:508]: Yeah

**Review by reaperhulk — APPROVED** [review]
